src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+            final FeatureContext context ) {
+        if( featureInputs == null ) { throw new IllegalArgumentException("featureInputs cannot be null."); }
+        if( context == null ) { throw new IllegalArgumentException("context cannot be null."); }
+
+        final ExpandingArrayList<VariantDatum> variants = new ExpandingArrayList<>();
+
+        for( final VariantContext vc : context.getValues(featureInputs, context.getInterval().getStart()) ) {
+            if( vc != null && ( IGNORE_ALL_FILTERS || vc.isNotFiltered() || ignoreInputFilterSet.containsAll(vc.getFilters()) ) ) {
+                if( VariantDataManager.checkVariationClass( vc, VRAC.MODE ) && !VRAC.useASannotations) {
+                    addDatum(variants, isInput, context, vc, null, null);
+                }
+                else if(VRAC.useASannotations) {
+                    for(final Allele allele : vc.getAlternateAlleles()) {
+                        if(allele == Allele.SPAN_DEL)
+                            continue;
+                        if (VariantDataManager.checkVariationClass(vc, allele, VRAC.MODE))
+                            addDatum(variants, isInput, context, vc, vc.getReference(), allele);
+                    }
+                }
+            }
+        }
+
+        return variants;
+    }
+
+    /**
+     * add a datum representing a variant site (or allele) to the data in {@code variants}, which represents the callset to be recalibrated
+     * @param variants is modified by having a new VariantDatum added to it
+     */
+    private void addDatum(
+            final ExpandingArrayList<VariantDatum> variants,
+            final boolean isInput,
+            final FeatureContext featureContext,
+            final VariantContext vc,
+            final Allele refAllele,
+            final Allele altAllele) {
+        final VariantDatum datum = new VariantDatum();
+
+        // Populate the datum with lots of fields from the VariantContext, unfortunately the VC is too big so we just
+        // pull in only the things we absolutely need.
+        datum.referenceAllele = refAllele;
+        datum.alternateAllele = altAllele;
+        dataManager.decodeAnnotations(datum, vc, true); //BUGBUG: when run with HierarchicalMicroScheduler this is
+
+        // non-deterministic because order of calls depends on load of machine
+        datum.loc = (isInput ? new SimpleInterval(vc.getContig(), vc.getStart(), vc.getEnd()) : null);
+
+        datum.originalQual = vc.getPhredScaledQual();
+        datum.isSNP = vc.isSNP() && vc.isBiallelic();
+        datum.isTransition = datum.isSNP && GATKVariantContextUtils.isTransition(vc);
+        datum.isAggregate = !isInput;
+
+        // Loop through the training data sets and if they overlap this locus (and allele, if applicable) then update
+        // the prior and training status appropriately
+        dataManager.parseTrainingSets(featureContext, featureContext.getInterval(), vc, datum, TRUST_ALL_POLYMORPHIC);
```

Comment:
This is used to query variants the overlap this location and start at the same position, so I think this is equivalent. Having sad that, I don't see anything in GATK3 that constrains this be a single-base location (getLocation returns a GenomeLoc, which has a start and a stop). It seems like it would be useful to have an explicit (type-level) distinction between a single locus and an interval to make usages like this less ambiguous, if it doesn't already exist.